libavutil/hwcontext_vaapi.c

```diff
@@ -1788,6 +1800,65 @@ static int vaapi_device_create(AVHWDeviceContext *ctx, const char *device,
     }
 #endif
 
+#if HAVE_VAAPI_WIN32
+    if (!display && try_win32) {
+        // Try to create a display from the specified device, if any.
+        if (!device) {
+            display = vaGetDisplayWin32(NULL);
+        } else {
+            IDXGIFactory2 *pDXGIFactory = NULL;
+            IDXGIAdapter *pAdapter = NULL;
+#if !HAVE_UWP
+            HANDLE dxgi = dlopen("dxgi.dll", 0);
+            if (!dxgi) {
+                av_log(ctx, AV_LOG_ERROR, "Failed to load dxgi.dll\n");
+                return AVERROR_UNKNOWN;
+            }
+            PFN_CREATE_DXGI_FACTORY pfnCreateDXGIFactory =
+                (PFN_CREATE_DXGI_FACTORY)dlsym(dxgi, "CreateDXGIFactory");
+            if (!pfnCreateDXGIFactory) {
+                av_log(ctx, AV_LOG_ERROR, "CreateDXGIFactory load failed\n");
+                return AVERROR_UNKNOWN;
+            }
+#else
+            // In UWP (which lacks LoadLibrary), CreateDXGIFactory isn't
+            // available, only CreateDXGIFactory1
+            PFN_CREATE_DXGI_FACTORY pfnCreateDXGIFactory =
+                (PFN_CREATE_DXGI_FACTORY)CreateDXGIFactory1;
+#endif
+            if (SUCCEEDED(pfnCreateDXGIFactory(&IID_IDXGIFactory2,
+                                              (void **)&pDXGIFactory))) {
+                int adapter = atoi(device);
+                if (SUCCEEDED(IDXGIFactory2_EnumAdapters(pDXGIFactory,
+                                                         adapter,
+                                                         &pAdapter))) {
+                    DXGI_ADAPTER_DESC desc;
+                    if (SUCCEEDED(IDXGIAdapter2_GetDesc(pAdapter, &desc))) {
+                        av_log(ctx, AV_LOG_INFO,
+                              "Using device %04x:%04x (%ls).\n",
+                              desc.VendorId, desc.DeviceId, desc.Description);
```

Comment:
vendor, device and description are not unique identifiers of the adapter in case of the system with multiple adapters (of the same type). I suggest that we need to also print LUID.